protocol/git/README.md

```diff
@@ -0,0 +1,77 @@
+# Git Protocol
+
+A guide for programming within version control.
+
+Maintain a Repo
+---------------
+
+* Use [GitFlow] as a branch management strategy.
+* Avoid including files in source control that are specific to your
+  development machine or process.
+* Delete local and remote feature branches after merging.
+* Perform work in a feature branch.
+* Rebase frequently to incorporate upstream changes.
+* Use a [pull request] for code reviews.
+
+[GitFlow]: https://nvie.com/posts/a-successful-git-branching-model/
+[pull request]: https://help.github.com/articles/using-pull-requests/
+
+Write a Feature
+---------------
+
+Create a local feature branch based off `develop`.
+
+    git checkout develop
+    git pull
+    git checkout -b <branch-name>
+
+Merge frequently to incorporate upstream changes.
+
+    git fetch origin
+    git merge origin/develop
```

Comment:
The Thoughtbot protocols that inspired this included instructions for rebasing, rather than merging, and for squashing commits to leave a cleaner, more digestible history. The downside to this, in my mind, is that those strategies require _not_ pushing work in progress up to the server, since they change history. If we're doing small enough feature branches, that's much less of a risk, and the cleaner history may be more valuable. I'd be interested if any of us have strong feelings one way or the other about this.

Reply:
Some good thoughts on the topic: https://medium.com/datadriveninvestor/git-rebase-vs-merge-cc5199edd77c

Reply:
Not pushing up work in progress up and altering commit history both go against the point of source control in my humble opinion.

Reply:
lgtm

Reply:
i've messed up a rebase once or twice and have avoided it since, but I can understand a few situations where it's valuable. Cool and thoughtful guide, I like it!

Reply:
I've always preferred merges since, as Stephen mentioned, it's just too easy to create a mess with rebasing. In general, I'm more on the side of committing more often and creating daily (nightly) builds than trying to achieve perfection in the commit history.